pkg/controllers/plugin/util.go

```diff
@@ -88,12 +94,26 @@ func initClientGetter(
 		return clusterAccessReadyCondition, restClientGetter
 	}
 
-	cluster := new(greenhousev1alpha1.Cluster)
-	err = k8sClient.Get(ctx, types.NamespacedName{Namespace: plugin.Namespace, Name: plugin.Spec.ClusterName}, cluster)
-	if err != nil {
-		clusterAccessReadyCondition.Status = metav1.ConditionFalse
-		clusterAccessReadyCondition.Message = fmt.Sprintf("Failed to get cluster %s: %s", plugin.Spec.ClusterName, err.Error())
-		return clusterAccessReadyCondition, nil
+	// only early exit if the plugin is not explicitly marked for deletion
+	if plugin.GetDeletionTimestamp() == nil {
+		// we early exit if the cluster is marked for deletion
+		logger := ctrl.LoggerFrom(ctx)
```

Comment:
Should we really mark a cluster as not `Ready` if it is scheduled for deletion? This would mean, the UI marks it as `erroring`. I would rather only reflect in `ClusterDeletionScheduled` Condition?

Reply:
Cluster is not marked but it is actually the Plugin `Ready` condition that is marked as not ready. I think this is fine because the `Cluster` is scheduled to be deleted and this reflects the `Plugins` are not being reconciled. It is part of the Condition message.

Reply:
The UI in the future should be adjusted to look for deletion marker annotations and not have it look like it is erroring

Reply:
@uwe-mayer - very good point instead of manipulating the `ClusterAccessReady` condition, we can set the same ClsuterDeletionScheduled condition on the plugin

```yaml
      - lastTransitionTime: '2024-10-02T12:30:16Z'
        message: ready
        status: 'True'
        type: Ready
      - lastTransitionTime: '2024-10-02T12:30:16Z'
        status: 'True'
        type: ClusterAccessReady
        .....
        message: Workload is running
        status: 'True'
        type: WorkloadReady
      - lastTransitionTime: '2024-10-02T12:31:08Z'
        message: >-
          cluster kind-greenhouse-remote is scheduled for deletion at 2024-10-04
          12:31:08 +0000 UTC
        status: 'True'
        type: ClusterDeletionScheduled
```

This will also simplify a lot of logic, thanks!